### DISCUSSION TITLE
Netbeans-based RCP, Open Project Dialog, missing folders on "Look in" ComboBox

### DISCUSSION THREAD

**Discussion by dmqvt** [apache/netbeans#4862] [Q&A] [answered]
I don't know whether this is a bug or just that "I'm not using it right".

I'm using Windows 10 and working on a complex RCP application. After migrating to Netbeans 13 and jdk17, the Open Project dialog looks different. The "Look in" ComboBox now shows only one item: "Desktop", there are no items hanging from it:
![grafik](https://user-images.githubusercontent.com/116723943/198021539-4cc9b303-71a4-4b27-8afa-8afdc82277f8.png)

Previously (Netbeans 8 and jdk 8) it showed a full tree, with "This PC" hanging from Desktop, and from it all drives, etc..
Using the Open Project on the Netbeans 13 IDE, the ComboBox also shows a full tree (as expected):
![grafik](https://user-images.githubusercontent.com/116723943/198020460-acbf38df-5417-4960-9934-2af7c1829285.png)

There is another change that may or not be related: To the left hand side, inside of the dialog, some icons were shown, but no longer in Netbeans 13. They are also missing when using the IDE:

![grafik](https://user-images.githubusercontent.com/116723943/198016374-bf47adb0-d033-4848-95b4-22f296b7c67b.png)
Previously:
![grafik](https://user-images.githubusercontent.com/116723943/198018984-4748cdb0-be4b-4e9d-b56c-a9cb62502939.png)

I've updated all the dependencies of the project to Netbeans 13, and the jdk to 17. This was a long process, but there was no change to anything related to the way the custom projects of the RCP are opened, There is no custom code for it apart from defining the Project types, but that only comes into play after the dialog is fully shown. There is no explicit reference to the "Open Project" dialog or actions anywhere in the layer files. 

I guess that something has changed (or has been fixed) in the newer versions of the Netbeans modules, but I've exhausted google and found no reference to this anywhere.

Is there a way to get back the previous tree behaviour? Even any background info about how this dialog works or how can it be configured/influenced would be very much appreciated.

Thanks!

**Comment by neilcsmith-net** [comment, marked as answer]
Which build system are you using?  This is possibly due to missing add-opens and add-exports in the configuration file necessary to run on JDK 17.

You could copy those from `etc/netbeans.conf` although might be more than you actually need.

**Reply by dmqvt** [reply]
Maven 3.8.3

Thanks for the reply and thanks a lot for the tip!

I drilled it down to:
-J--add-opens=java.desktop/sun.awt.shell=ALL-UNNAMED

That enables the full tree in jdk17. 
Even when using all the default config from the Netbeans IDE I haven't been able to reactivate the icons on the left of the dialog. They are not visible in the Netbeans 13 IDE so I'm assuming they have been removed on purpose. That's fine with the RCP App I'm working on.

Thanks again!
